src/config.rs

```diff
@@ -193,52 +165,40 @@ impl Config {
         query: String,
         poll_interval: i32,
     ) {
-        match &mut self.sources {
-            Some(sources) => sources.push(Source::postgres_connector {
-                display_name,
-                url,
-                origin,
-                query,
-                poll_interval,
-            }),
-            None => {
-                self.sources = Some(vec![Source::postgres_connector {
-                    display_name,
-                    url,
-                    origin,
-                    query,
-                    poll_interval,
-                }])
-            }
-        }
+        self.get_mut_sources().push(Source::postgres_connector {
+            display_name,
+            url,
+            origin,
+            query,
+            poll_interval,
+        });
     }
 
     pub(crate) fn add_mysql_connector_source(
         &mut self,
         display_name: String,
         url: String,
-        schema: String,
-        tables: String,
+        origin: String,
+        query: String,
         poll_interval: i32,
     ) {
-        match &mut self.sources {
-            Some(sources) => sources.push(Source::mysql_connector {
-                display_name,
-                url,
-                schema,
-                tables,
-                poll_interval,
-            }),
-            None => {
-                self.sources = Some(vec![Source::mysql_connector {
-                    display_name,
-                    url,
-                    schema,
-                    tables,
-                    poll_interval,
-                }])
-            }
-        }
+        self.get_mut_sources().push(Source::mysql_connector {
+            display_name,
+            url,
+            origin,
+            query,
+            poll_interval,
+        })
+    }
+
+    fn get_mut_destinations(&mut self) -> &mut Vec<Destination> {
```

Comment:
Non blocking note: the pattern for this seems to be repeating `get_mut_destinations.push(...)`, so that could be shortened with something like `add_destination(...)`

Reply:
That looks good, thank you!